otlp/otlptraces/traces.go

```diff
@@ -109,6 +110,11 @@ func TracesModule(cfg ModuleConfig) fx.Option {
 						return otlptracegrpc.WithInsecure()
 					}))
 				}
+				if cfg.OTLPConfig.Headers != nil {
```

Comment:
```suggestion
				if len(cfg.OTLPConfig.Headers) > 0 {
```

Since you use `make` on the map whether or not you append any headers a length check would be more appropriate.